packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx

```diff
@@ -0,0 +1,158 @@
+import { useContext, useState } from "react";
+import type { ThirdwebClient } from "../../../client/client.js";
+import type { PreparedTransaction } from "../../../transaction/prepare-transaction.js";
+import { useSendTransactionCore } from "../../core/hooks/contract/useSendTransaction.js";
+import { SetRootElementContext } from "../../core/providers/RootElementContext.js";
+import {
+  type SupportedTokens,
+  defaultTokens,
+} from "../ui/ConnectWallet/defaultTokens.js";
+import { useConnectLocale } from "../ui/ConnectWallet/locale/getConnectLocale.js";
+import { BuyScreen } from "../ui/ConnectWallet/screens/Buy/SwapScreen.js";
+import { SwapTransactionsScreen } from "../ui/ConnectWallet/screens/SwapTransactionsScreen.js";
+import { Modal } from "../ui/components/Modal.js";
+import { CustomThemeProvider } from "../ui/design-system/CustomThemeProvider.js";
+import type { Theme } from "../ui/design-system/index.js";
+import type { LocaleId } from "../ui/types.js";
+import { LoadingScreen } from "../wallets/shared/LoadingScreen.js";
+
+/**
+ * Configuration for the `useSendTransaction` hook.
+ */
+export type SendTransactionConfig = {
+  /**
+   * Configuration for the "Buy Modal" that opens when the user doesn't have enough funds to send a transaction.
+   *
+   * This configuration object includes the following properties to configure the "Buy Modal" UI:
+   *
+   *
+   * ### `locale`
+   * The language to use for the "Buy Modal" UI. Defaults to `"en_US"`.
+   *
+   * ### `supportedTokens`
+   * An object of type [`SupportedTokens`](https://portal.thirdweb.com/references/typescript/v5/SupportedTokens) to configure the tokens to show for a chain.
+   *
+   * ### `theme`
+   * The theme to use for the "Buy Modal" UI. Defaults to `"dark"`.
+   *
+   * It can be set to `"light"` or `"dark"` or an object of type [`Theme`](https://portal.thirdweb.com/references/typescript/v5/Theme) for a custom theme.
+   *
+   * Refer to [`lightTheme`](https://portal.thirdweb.com/references/typescript/v5/lightTheme)
+   * or [`darkTheme`](https://portal.thirdweb.com/references/typescript/v5/darkTheme) helper functions to use the default light or dark theme and customize it.
+   */
+  buyModal?: {
```

Comment:
add option `false` to explicitly disable the buyModal behavior

Reply:
done